test_project/test_utils/dbt_test_utils.py

```diff
@@ -383,22 +419,75 @@ def context_table_to_csv(self, table: Table, model_name: str) -> str:
 
         return csv_fqn.stem
 
-    def context_table_to_dict(self, table: Table, orient='index'):
+    def context_table_to_dict(self, table: Table, orient='index', use_nan=True):
         """
         Converts a context table in a feature file into a dictionary
             :param table: The context.table from a scenario
             :param orient: orient for df to_dict
+            :param use_nan: Replace <null> placeholder with NaN
             :return: A dictionary modelled from a context table
         """
 
-        table_df = self.context_table_to_df(table)
+        table_df = self.context_table_to_df(table, use_nan=use_nan)
 
         table_dict = table_df.to_dict(orient=orient)
 
         table_dict = self.parse_lists_in_dicts(table_dict)
 
         return table_dict
 
+    def context_table_to_model(self, seed_config : dict, table: Table, model_name: str, target_model_name: str):
```

Comment:
What is the purpose of this function? We can already do this with calls to a number of smaller functions, which is much cleaner. Let's discuss

Reply:
Function context_table_to_model() creates a table from context / expected data while staying in control of datatypes and null placeholders. dbt's default processing of context / expected data and datatypes does not work with MSSQL at all.
Another key feature of this function is that it creates the target seed table with a name in lower case, whereas MSSQL specific code elsewhere creates target actual tables with names in upper case. In the end this was the only way I could find to ensure that dbt did not trip itself up when 'clearing the decks' at the start of a test run and during initial table populations (from seed to target tables) when it would erroneously report the existence of duplicate tables on MSSQL. 
All tested successfully on Snowflake too.